project-chip/zap
# Adding tests for Matter custom xml support

Added tests for **Matter custom xml support** for the following use cases:
- loads a file with a custom cluster and an extension of a cluster and checks it
- loads a second file with a further extension of the extended cluster and checks it
- tests generation with custom xml
- loading bad custom xml -> throw warning
- Add 2 custom files with manufacturer name/code and attribute code collisions -> throw error.
    - This should be an SQL error, it is currently broken and should be fixed in PR #1352 is merged.
- test generation after deleting the custom xml
    - This doesn't work as expected - <https://github.com/project-chip/zap/issues/1387>
- Add custom xml, make changes, delete, add again -> make sure that generated content has new edits
     - Effected by <https://github.com/project-chip/zap/issues/1387> and <https://github.com/project-chip/zap/issues/1391>
- Loading zap file with custom xml that does not exist - throw error
- Loading zap file with custom xml and generating

Once PR #1352 is merged and #1387 and #1391 are resolved - tests should be updated appropriately.
